packages/scaffold-ui/src/views/w3m-connect-view/index.ts

```diff
@@ -49,15 +51,18 @@ export class W3mConnectView extends LitElement {
 
   public override disconnectedCallback() {
     this.unsubscribe.forEach(unsubscribe => unsubscribe())
-    const connectEl = this.shadowRoot?.querySelector('.connect')
-    connectEl?.removeEventListener('scroll', this.handleConnectListScroll.bind(this))
+    this.resizeObserver?.disconnect()
   }
 
   public override firstUpdated() {
     const connectEl = this.shadowRoot?.querySelector('.connect')
-    // Use requestAnimationFrame to access scroll properties before the next repaint
-    requestAnimationFrame(this.handleConnectListScroll.bind(this))
-    connectEl?.addEventListener('scroll', this.handleConnectListScroll.bind(this))
+    if (connectEl) {
+      this.resizeObserver = new ResizeObserver(() => {
+        this.handleConnectListScroll()
+      })
+      this.resizeObserver.observe(connectEl)
```

Comment:
do we need to unbind this at any point?

Reply:
`this.resizeObserver?.disconnect()` does the thing in `disconnectedCallback`